target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -18,12 +18,19 @@ ath10kcal_extract() {
 	local part=$1
 	local offset=$2
 	local count=$3
-	local mtd
+	local mtd size1 size2
 
 	mtd=$(find_mtd_chardev $part)
 	[ -n "$mtd" ] || \
 		ath10kcal_die "no mtd device found for partition $part"
 
+	# Check that the calibration data size in header equals the desired size
+	size1=`dd if=$mtd bs=2 count=1 skip=$(( $offset / 2 )) conv=swab 2>/dev/null | hexdump -ve '1/2 "%.4x"'`
```

Comment:
Please use var=`$(...)` instead of ``var=`...` ``. And, what about something different (no need to change count to hex):

`cal_size=$(dd if=$mtd bs=2 count=1 skip=$(($offset/2)) conv=swab 2>/dev/null | hexdump -ve '1/2 "%d"')`

Reply:
Heh, I was somehow so completely fixated on getting it to hex that the easier route eluded me. I'll send a new version soon.